splash/network_manager.py

```diff
@@ -226,7 +240,12 @@ def _handle_custom_headers(self, request):
             headers = headers.items()
 
         for name, value in headers or []:
-            request.setRawHeader(to_bytes(name), to_bytes(value))
+            try:
+                request.setRawHeader(to_bytes(name), to_bytes(value))
+            except TypeError:
+                msg = "invalid header {}: {}. Header keys and values must be string or bytes"
+                self.log(msg.format(name, value), min_level=1, format_msg=False)
+                continue
```

Comment:
I think that's fine to accept numbers as header values without raising an error; it looks like a common pain point.